Marlin/ultralcd_impl_DOGM.h

```diff
@@ -183,9 +183,13 @@
   //U8GLIB_SSD1306_128X64 u8g(U8G_I2C_OPT_NONE | U8G_I2C_OPT_FAST);  // 8 stripes
   U8GLIB_SSD1306_128X64_2X u8g(U8G_I2C_OPT_NONE | U8G_I2C_OPT_FAST); // 4 stripes
 #elif ENABLED(U8GLIB_SH1106)
+  #if ENABLED(MKS_12864OLED)
+    U8GLIB_SH1106_128X64 u8g(23, 17, 16, 25); // SW SPI Com: SCK = 23, MOSI = 17, CS = 16, A0 = 25
```

Comment:
What's with the hard-coded pin numbers? According to your addition to the `pins_RAMPS.h` file, we end up with these pins:
```cpp
    #if ENABLED(MKS_12864OLED)
      #define LCD_PINS_DC  25
      #define LCD_PINS_RST 27
    #else
      #define LCD_PINS_D5  25
      #define LCD_PINS_D6  27
    #endif

    #define LCD_PINS_RS    16
    #define LCD_PINS_ENABLE 17
    #define LCD_PINS_D4    23
    #define LCD_PINS_D7    29
```
So this line might be written instead as:
```cpp
U8GLIB_SH1106_128X64 u8g(LCD_PINS_D4, LCD_PINS_ENABLE, LCD_PINS_RS, LCD_PINS_DC);
```
Was that your intention?

Reply:
I am not familiar with your functions and macros.  That was the OEM recommendation.  If you show me the final code, I'll try it.

Reply:
ok, i understand, now.  I'll try it

Reply:
that worked